internal/impl/devbox.go

```diff
@@ -1190,3 +1176,34 @@ func (d *Devbox) addHashToEnv(env map[string]string) error {
 	}
 	return err
 }
+
+func (d *Devbox) convertEnvToMap(currentEnv []string) (map[string]string, error) {
+	env := make(map[string]string, len(currentEnv))
+	for _, kv := range currentEnv {
+		key, val, found := strings.Cut(kv, "=")
+		if !found {
+			return nil, errors.Errorf("expected \"=\" in keyval: %s", kv)
+		}
+		if ignoreCurrentEnvVar[key] {
+			continue
+		}
+		// handling special cases to for pure shell
+		// Passing HOME USER and DISTPLAY for pure shell to leak through
+		// otherwise devbox binary won't work - this matches nix
+		if !d.pure || key == "HOME" || key == "USER" || key == "DISPLAY" || key == "PATH" {
```

Comment:
What's the motivation for including `PATH` even in `--pure` mode?

PATH seems to be one of the critical env-vars that we definitely want to omit from the external environment for pure mode, because then all sorts of external programs become callable inside the pure shell.

Reply:
Oh I see. Could you add a small comment like:
```
// We include PATH to find the nix installation. It is cleaned for pure mode below.
```
?